exo-go/src/docker_helpers/index.go

```diff
@@ -17,9 +18,60 @@ func CatFileInDockerImage(c *client.Client, image, fileName string) ([]byte, err
 	return []byte(output), err
 }
 
+// ListRunningContainers passes a slice of the names of running containers
+// and error (if any) to the callback function
+func ListRunningContainers(c *client.Client, done func([]string, error)) {
```

Comment:
I think we need to cleanup the running containers after each step then or do better filtering in the step. The callback shouldn't change anything aside from preventing that step from being able to return an error